### ISSUE TITLE
[MU4 Issue] Key signature preferences for transposing instruments affecting concert pitch key signature

### ISSUE THREAD

**Issue by mccoydtromb** [musescore/MuseScore#14274]
In Stave/Part properties, when "Prefer sharps or flats for transposed key signatures" is set to Flats, that instrument is given flats when the score is in Concert pitch also, not just in transposed pitch.

**To Reproduce**
1. Create a score with Trumpet and Piano (for example) in the key of B major.
2. Right click trumpet stave and go to Stave/Part Properties
3. Select Flats in "Prefer sharps or flats for transposed key signatures"
4. Notice trumpet is now in D flat major (expected behaviour)
5. Toggle Concert Pitch
6. Notice trumpet is now in C flat major.

**Expected behavior**
I would expect the Concert Pitch score to have the same key signature across all instruments, not work with enharmonics.

![image](https://user-images.githubusercontent.com/92689427/199567569-f2a73810-7cfa-4cee-a913-8e2063a31cc2.png)

**Platform information**
 - OS: Windows 10
 - MuseScore: MU4 Beta

Found by a colleague.
Found a similar problem when Sharps is selected.

**Comment by cbjeukendrup** [comment]
This also seems to occur in MS 3.6.2.

**Comment by Tantacrul** [comment]
Yes, this should be moved to 4.x. @oktophonie - any thoughts? Makes sense to me that at concert pitch, the 'prefer flats/sharps' setting should be ignored.

**Comment by sammik** [comment]
Also, if I have "prefer sharps or flats - default" and at concert pitch have F# major and switch to transposed pitch - it shows Ab major (expected), but switching back to concert pitch it transposes to Gb.


https://user-images.githubusercontent.com/1646034/200684689-f3554c29-1920-400e-8de8-50c79a094cee.mp4

So probably "prefer flat / sharp" needs to have two values, one for transposed and one for concert. And option in "staff / part properties" should affect (set) only transposed value (like Tantacrul said)
Concert value should be set by key signature - if it is sharp key signature to "prefer sharps" and if it is flat key signature to "prefer flat".

**Comment by oktophonie** [comment]
A not-uncommon situation is where a piece in (for example) B major would have a harp part notated in C flat major. This could be achieved with a local key signature for the harp (as it's may not be desired consistently, even within the same score); and since the option is specifically called "Prefer sharps or flats for transposed key signatures", it makes sense to me for it to have no effect in a concert pitch score.

**Comment by sammik** [comment]
Couple of trhings to think about.

1) Rule: "less accidentals is better"
If score is B major (5#), than Bb clarinet should be C# major (7#), or better Db major (5b), like f.ex. Dorico does. 
Muse Score:
![MU-transposed pitch](https://user-images.githubusercontent.com/1646034/224251791-945313ff-72ee-4b8a-bb7d-a07db939e0ad.png)
Dorico:
![Dorico](https://user-images.githubusercontent.com/1646034/224251828-0578b7fd-8a45-442c-a0b6-130acde3f32f.png)

2) Also notes should reflect key signature in transposed pitch, so not mechanicaly transpose instrument (+2), but take key signtarue into account
Now Muse Score does it wrong:
![Bug-notes](https://user-images.githubusercontent.com/1646034/224252229-0c1363f3-b61f-4283-b806-c1d7ad0096f0.png)

**Comment by sammik** [comment]
Other tip in my mind.

May be better, to make this option "prefer flats / sharps" individual for each KeySignature, instead of whole Staff.

**Comment by sammik** [comment]
Ill try to fix this.
